tests/test_coupons.py

```diff
@@ -122,6 +123,17 @@ def test_delete(self):
             coupon.refresh_from_db()
             self.assertTrue(coupon.is_deleted)
 
+        with mock.patch("aa_stripe.models.StripeCoupon.delete") as mocked_delete:
```

Comment:
The mock used here is incorrect for deleted coupons.
Please update.
https://stripe.com/docs/api#delete_coupon